samples/update_protection_runs/update_job_run.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2019 Cohesity Inc.
+#
+# Python example to extend the snapshot retention per protection job run. The
+# runs can be filtered by start/end date.
+#
+# Usage : python update_job_run.py --job_name=<protect-job>
+# --start_time=2019-2-31 --end_time=2019-7-29 --extend=<num-of-days>
+#
+# --extend = 30 # Extends snapshot retention by 30 days or 1 month.
+# --extend = 0 # Deletes the snapshot.
+# --extend = -10 # Reduces the snapshot by 10 days.
+
+
+import argparse
+import datetime
+
+from cohesity_management_sdk.cohesity_client import CohesityClient
+from cohesity_management_sdk.models.type_run_job_snapshot_target_enum import TypeRunJobSnapshotTargetEnum
+from cohesity_management_sdk.models.run_job_snapshot_target import RunJobSnapshotTarget
+from cohesity_management_sdk.models.update_protection_job_run import UpdateProtectionJobRun
+from cohesity_management_sdk.models.update_protection_job_runs_param import UpdateProtectionJobRunsParam
+from cohesity_management_sdk.models.universal_id import UniversalId
+
+CLUSTER_VIP = 'CLUSTER_VIP'
+CLUSTER_USERNAME = 'CLUSTER_USERNAME'
+CLUSTER_PASSWORD = 'CLUSTER_PASSWORD'
+DOMAIN = 'LOCAL'
+
+MILLISECONDS = '0' * 6
+cohesity_client = CohesityClient(cluster_vip=CLUSTER_VIP,
+                                 username=CLUSTER_USERNAME,
+                                 password=CLUSTER_PASSWORD,
+                                 domain=DOMAIN)
+def get_pro_id(job_name):
+    """
+    Function to get job related attributes
+    :param job_name:
+    :return: job_id, cluster_id and cluster_incarnation_id.
+    """
+
+    job = cohesity_client.protection_jobs.get_protection_jobs(names=job_name)[0]
+    return job.uid.id, job.uid.cluster_id, job.uid.cluster_incarnation_id
+
+def convert_to_epoch(date, end_time=False):
+    """
+    Helper function to convert date to epochtime
+    :param date:
+    :return: datetime
+    """
+    if end_time:
+        # This is to get the time till end of the day.
+        return datetime.datetime(int(date[0]), int(date[1]),
+                                 int(date[2]), 23, 59).strftime('%s')
+    return datetime.datetime(int(date[0]), int(date[1]), int(date[2])).strftime('%s')
+
+def convert_to_date(epoch):
+    """
+    Convert epoch to date
+    :param epoch:
+    :return:
+    """
+    return datetime.datetime.fromtimestamp(epoch/1000000).strftime('%c')
+
+def change_retention(job_run, job_id, cluster_id, incarnation_id, extend):
+    """
+    Change the snapshot retention time.
+    :param job_run:
+    :param job_id:
+    :param cluster_id:
+    :param incarnation_id:
+    :param extend:
+    :return:
+    """
+    body = UpdateProtectionJobRunsParam()
+    body.job_runs = []
+    index = 0
+
+    for run in job_run:
```

Comment:
Check for the empty response for job_run

Reply:
Good catch!